src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -1273,7 +1276,7 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
 
     const settings = stream.getTracks()[0].getSettings();
     // If a device does not specify deviceId, we have to assume the stream is not reusable.
-    return settings.deviceId === this.getIntrinsicDeviceIdStr(device);
+    return settings.deviceId === DefaultDeviceController.getIntrinsicDeviceIdStr(device);
```

Comment:
There might have been a bug here!

Reply:
Add a check to make sure settings.deviceId is not undefined, null, or ''.